library/net.cpp

```diff
@@ -1204,23 +1203,34 @@ static net_state_list_ptr dnet_check_route_table_victims(struct dnet_node *node,
 
 static net_addr_list dnet_reconnect_victims(struct dnet_node *node, int *flags)
 {
-	net_addr_list addrs;
-
 	dnet_pthread_lock_guard locker(node->reconnect_lock);
 
-	addrs.reserve(node->reconnect_num);
+	if (!node->reconnect_num)
+		return net_addr_list();
+
+	size_t addresses_needed = std::min(node->reconnect_parallelism, size_t(node->reconnect_num));
+
+	net_addr_list addrs;
+	addrs.reserve(addresses_needed);
 
+	size_t addresses_collected = 0;
```

Comment:
`addresses_collected `  is always equal to `addrs.size()`, isn't it?

Reply:
Thanks, you're right. I lost this while algorithm evolution.